### DISCUSSION TITLE
How to include CHGCAR and WAVECAR  et al. as input files in the 02.fp  vasp steps?

### DISCUSSION THREAD

**Discussion by liupf16** [deepmodeling/dpgen#1172] [Q&A] [answered]
In the source code, it says
     "optional_files: Other input files supplied as a dict of { filename: object}. The object should follow standard pymatgen conventions in implementing a as_dict() and from_dict method." When I set this parameter as follows:
    **"optional_files":{'WAVECAR':WAVECAR},
    "fp_incar": "./INCAR_scf"**
It does't work.

**Comment by AnguseZhang** [comment, marked as answer]
You may refer to https://github.com/deepmodeling/dpgen/discussions/436.  Search "user_forward_files" in documentation. Hope this works.

**Reply by liupf16** [reply]
The problem has been solved. I found a related template in the link：
https://github.com/deepmodeling/dpgen/blob/master/examples/CH4-refact-dpdispatcher/machine-ali-ehpc.json
Thany you very much
